dapps/shop/backend/queues/makeOfferProcessor.js

```diff
@@ -0,0 +1,127 @@
+const queues = require('./queues')
+
+const Web3 = require('web3')
+
+const { ListingID } = require('../utils/id')
+const { Shop, Network } = require('../models')
+const { post, getBytes32FromIpfsHash } = require('../utils/_ipfs')
+const encConf = require('../utils/encryptedConfig')
+const abi = require('../utils/_abi')
+
+const ZeroAddress = '0x0000000000000000000000000000000000000000'
+
+function attachToQueue() {
+  const queue = queues['makeOfferQueue']
+  queue.process(processor)
+  queue.resume() // Start if paused
+}
+
+/**
+ * Processes a credit card transaction and submit it to the blockchain.
+ *
+ * job.data should have {shopId, amount, encryptedData}
+ * @param {*} job
+ */
+async function processor(job) {
```

Comment:
Might recommend using `job.progress` to show completion.  It can sometimes help with troubleshooting, though perhaps `job.log` takes care of that here.